doc/banner/banner_hld.md

```diff
@@ -0,0 +1,263 @@
+
+# Scope
+The scope of this document is to cover definition, design and implementation of SONiC Banner feature and Banner CLI.
+The document covers the next CLI:
+1.	Commands to configure banner messages
+2.	Command to display banner settings
+
+<!-- omit in toc -->
+## Abbreviations
+| Term  | Meaning                                   |
+|:------|:------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+| MOTD  | Message of the day                        |
+| DB    | Database                                  |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+<!-- omit in toc -->
+## List of figures
+- [Figure 1: Banner system chart diagram](#figure-1-banner-system-chart-diagram)
+- [Figure 2: Banner init flow](#figure-2-banner-init-flow)
+- [Figure 3: Banner config flow](#figure-3-banner-config-flow)
+- [Figure 4: Banner show configuration](#figure-4-banner-show-configuration)
+
+# 1 Introduction
+
+## 1.1 Feature overview
+The SONiC maintains several messages for communication with users. These messages are associated with login and logout processes.
+
+There are few banners message types:
+|  Type         |  Description                                                                    |
+| :-----------: | :-----------------------------------------------------------------------------: |
+|  login        | Display a banner to the users connected locally or remotely before login prompt |
+|  motd         | Display a banner to the users after login prompt                                |
+|  logout       | Display a logout banner to the users connected locally or remotely              |
+
+
+## 1.2 Requirements
+
+**This feature will support the following functionality:**
+1. Show Banner configuration
+2. Configure Banner
+    1. Feature state
+    2. Login message
+    3. Logout message
+    4. MOTD
+
+# 2 Design
+
+## 2.1 Overview
+Here is the representation of SONiC platform using Banner feature:
+
+![Banner system chart diagram](images/banner_system_chart_diagram.png "Figure 1: Banner system chart diagram")
+<!-- omit in toc -->
+###### Figure 1: Banner system chart diagram
+
+This feature require access to SONiC DB. All messages (MOTD, login and logout) saved into the SONiC config database. Hostcfgd will listen for the configuration changes in corresponding tables and restart banner-config service. Banner config service - it is simple SystemD service which runs before we get SSH connection. It reads configured messages from database and apply it to Linux.
+
+## 2.2 Flows
+<!-- omit in toc -->
+### 2.2.1 Banner init flow
+![Banner init flow](images/banner_init_diagram.png "Figure 2: Banner init flow")
+<!-- omit in toc -->
+###### Figure 2: Banner init flow
+
+<!-- omit in toc -->
+### 2.5.2 Banner config flow
+![Banner config flow](images/banner_config_diagram.png "Figure 3: Banner config flow")
```

Comment:
I would recommend to update the banner files directly from hostcfgd service. This will avoid introducing another service (banner-config) which is intended to fetch the data from redis & update the banner files.
Is there any advantage of introducing new host service?

Reply:
The hostcfgd (after reboot) start later then network service - means banner might not be updated when user login to the system.
It  is main goal to use the systemd service as single point where the banners will be updated and it will start before network service.